applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_base_element.cpp

```diff
@@ -615,7 +617,10 @@ double AdjointFiniteDifferencingBaseElement::GetPerturbationSizeModificationFact
 
     if(rDesignVariable == SHAPE)
     {
-        KRATOS_ERROR << "GetPerturbationSizeModificationFactor NOT_IMPLEMENTED" << std::endl;
+        const double domain_size = mpPrimalElement->GetGeometry().DomainSize();
+        KRATOS_DEBUG_ERROR_IF(domain_size <= std::numeric_limits<double>::epsilon())
```

Comment:
What are you doing here?

Reply:
I ask the element for its size. This value is then later on multiplied with a given scaling factor. So lets imagine we have a beam element with a length of 0.5 and the scaling factor is 1e-5 than the final perturbation measure to compute shape derivatives would be 5e-6. We found that its good to set the perturbation measure in relation to the element size.
Before this PR this size was computed in the derived elements but we decided to have a default function in the adjoint base element to get this measure.

Reply:
@MFusseder can you add this explanation as a comment in the code pls?

Reply:
I added  a comment.

Reply:
Thanks!